homeassistant/components/apsystems/number.py

```diff
@@ -45,7 +47,13 @@ def __init__(
 
     async def async_update(self) -> None:
         """Set the state with the value fetched from the inverter."""
-        self._attr_native_value = await self._api.get_max_power()
+        try:
+            status = await self._api.get_max_power()
+        except ClientConnectorError:
+            self._attr_available = False
```

Comment:
See https://developers.home-assistant.io/docs/core/integration-quality-scale/rules/log-when-unavailable

> If internet/device/service is unavailable, log once when unavailable and once when back connected

It looks to me like you never log...

Reply:
But this would run on every update, wouldn't it?

Reply:
> But this would run on every update, wouldn't it?

Yes, but we would implement a variable to remember if we have already logged the availability change.

Reply:
> See https://developers.home-assistant.io/docs/core/integration-quality-scale/rules/log-when-unavailable
> 
> > If internet/device/service is unavailable, log once when unavailable and once when back connected
> 
> It looks to me like you never log...
> 
> Edit: Or maybe it's because you don't want to log for each entity...

This integration uses partly a DataUpdateCoordinator (https://developers.home-assistant.io/docs/integration_fetching_data?_highlight=fetching&_highlight=data#coordinated-single-api-poll-for-data-for-all-entities) and partly separate polling (https://developers.home-assistant.io/docs/integration_fetching_data?_highlight=fetching&_highlight=data#separate-polling-for-each-individual-entity).
With #131930 implemented we would log availability for the coordinator. Which should in most cases also apply for the single entities. Maybe we can raise UpdatedFailed in #131930 for some more Exceptions, to get more cases coverd?